donut/modules/arcfeedback/routes.py

```diff
@@ -0,0 +1,190 @@
+import flask
+from donut.modules.arcfeedback import blueprint
+from donut.modules.arcfeedback import helpers
+from donut import auth_utils
+from datetime import datetime
+from donut.modules.arcfeedback.permissions import arc_permissions
+from donut.default_permissions import Permissions as default_permissions
+
+
+@blueprint.route('/arcfeedback')
+def arcfeedback():
+    summary = auth_utils.check_login() and \
+              auth_utils.check_permission(flask.session['username'],
+                                          arc_permissions.SUMMARY)
+    return flask.render_template('arcfeedback.html', summary=summary)
+
+
+# submit feedback form
+@blueprint.route('/arcfeedback/submit', methods=['POST'])
+def arcfeedback_submit():
+    fields = ['name', 'email', 'course', 'msg']
+    required = ['course', 'msg']
+    max_len = {'name': 50, 'email': 50, 'course': 50, 'msg': 5000}
+    data = {}
+    for field in fields:
+        data[field] = flask.request.form.get(field)
+    for field in required:
+        if (data[field] == ""):
+            flask.flash('Please fill in all required fields (marked with *)',
+                        'error')
+            return flask.redirect(flask.url_for('arcfeedback.arcfeedback'))
+    for field in data:
+        if len(data[field]) > max_len[field]:
+            flask.flash(
+                'Error: field %s exceeded the maximum character length' %
+                (field))
+            return flask.redirect(flask.url_for('arcfeedback.arcfeedback'))
+    complaint_id = helpers.register_complaint(data)
+    if data['email'] != "":
+        helpers.send_confirmation_email(data['email'], complaint_id)
+    flask.flash('Success. Link: ' + helpers.get_link(complaint_id))
+    return flask.redirect(flask.url_for('arcfeedback.arcfeedback'))
+
+
+# api endpoint with all visible data
+@blueprint.route('/1/arcfeedback/view/<uuid:id>')
+def arcfeedback_api_view_complaint(id):
+    if not (helpers.get_id(id)):
+        return flask.render_template("404.html")
+    complaint_id = helpers.get_id(id)
+    #pack all the data we need into a dict
+    return flask.jsonify(helpers.get_all_fields(complaint_id))
+
+
+# view a complaint
+@blueprint.route('/arcfeedback/view/<uuid:id>')
+def arcfeedback_view_complaint(id):
+    if not (helpers.get_id(id)):
+        return flask.render_template("404.html")
+    complaint_id = helpers.get_id(id)
+    complaint = helpers.get_all_fields(complaint_id)
+    complaint['uuid'] = id
+    perms = set()
+    if auth_utils.check_login():
+        perms = auth_utils.get_permissions(flask.session['username'])
+    if default_permissions.ADMIN in perms:
+        arcperms = set(arc_permissions)
+    else:
+        arcperms = perms & (set(arc_permissions))
+    namedperms = [perm.name for perm in arcperms]
```

Comment:
auth_utils.get_permissions(flask.session['username']) only returns the permission_id ints not the whole enum so when arcperms = perms & (set(arc_permissions)) happens arcperms only contains ints and there's no name attribute for int
You can do
[arc_permissions(perm).name for perm in arcperms]

Reply:
Actually, when you take the intersection, it preserves the names from `arc_permissions` so it works as is